pkg/cvo/cvo.go

```diff
@@ -499,28 +499,54 @@ func (optr *Operator) clusterOperatorEventHandler() cache.ResourceEventHandler {
 	return cache.ResourceEventHandlerFuncs{
 		UpdateFunc: func(old, new interface{}) {
 			versionName := "operator"
-			_, oldVersion := clusterOperatorInterfaceVersionOrDie(old, versionName)
-			newStruct, newVersion := clusterOperatorInterfaceVersionOrDie(new, versionName)
+			oldStruct := clusterOperatorInterfaceStructOrDie(old)
+			newStruct := clusterOperatorInterfaceStructOrDie(new)
+
+			oldVersion := clusterOperatorVersion(oldStruct, versionName)
+			newVersion := clusterOperatorVersion(newStruct, versionName)
 			if optr.configSync != nil && oldVersion != newVersion {
 				msg := fmt.Sprintf("Cluster operator %s changed versions[name=%q] from %q to %q", newStruct.ObjectMeta.Name, versionName, oldVersion, newVersion)
 				optr.configSync.NotifyAboutManagedResourceActivity(new, msg)
+				return
+			}
+
+			oldAvailable := clusterOperatorConditionStatus(oldStruct, configv1.OperatorAvailable)
+			newAvailable := clusterOperatorConditionStatus(newStruct, configv1.OperatorAvailable)
+			if optr.configSync != nil && oldAvailable != newAvailable {
+				msg := fmt.Sprintf("Cluster operator %s changed Available from %q to %q", newStruct.ObjectMeta.Name, oldAvailable, newAvailable)
```

Comment:
nit: you could use `%s` and `configv1.OperatorAvailable` here to avoid hard-coding the `type` string, although it seems unlikely that it will ever diverge from `Available`.

Reply:
Done